activesupport/lib/active_support/core_ext/object/json.rb

```diff
@@ -51,6 +51,8 @@ def as_json(options = nil) #:nodoc:
     else
       instance_values.as_json(options)
     end
+  rescue SystemStackError
```

Comment:
This was my concern too. Is it too brittle to check the message? Or is the risk of having the `rescue` even worth it to provide better error feedback?